### PR TITLE
chore(fr-translation): update whole French translation (based on 1.3.0 release)

### LINKED ISSUE #16550: French translation not usable: contribution proposal
**French translation is not usable** (too many mistakes/missing translations)

For instance within `superset/superset/translations/fr/LC_MESSAGES/messages.json`

* `"Annotation delete failed.": ["Annotations et couches"]` -> it means "Annotations and layers"
* `"Annotation layer parameters are invalid.": ["Les couches d'annotation sont toujours en cours de chargement."]` -> it means "Annotation layers are still loading"

**Contribution proposal**

I've created this issue to avoid effort duplication by informing other potential contributors that I'll propose a PR for an improved French translation for Superset `1.3.0` during September 2021.

### PR DESCRIPTION
**Pull request by audour**
### SUMMARY
Based on [1.3.0 release](https://github.com/apache/superset/releases/tag/1.3.0), update whole French translation by following [CONTRIBUTING.md#translating](https://github.com/apache/superset/blob/master/CONTRIBUTING.md#translating).

### BEFORE/AFTER SCREENSHOTS

Before:
2021-09-28_before

After:
2021-09-28_after

### ADDITIONAL INFORMATION
- [x] Has associated issue: 
  - Fixes #16550
  - Issue reported during translation process: #16613
- followed process (based on [CONTRIBUTING.md#translating](https://github.com/apache/superset/blob/master/CONTRIBUTING.md#translating)):
  * extract new strings for translation (update the template file `superset/translations/messages.pot`)
  * update language files with the new extracted strings (update the file `superset/translations/fr/LC_MESSAGES/messages.po`)
  * review each translation: fix existing ones and add missing ones
  * convert PO file to formatted JSON file (update the file `superset/translations/fr/LC_MESSAGES/messages.json`)
  * compile PO file to MO file and test MO/JSON files on a running `1.3.0` superset instance
  * commit both `messages.po` and `messages.json`

### CODE REVIEW

**Review by mathieudruart — COMMENTED** [review]
I reviewed the first 1300 lines (I will continue later)

**Review by mistercrunch — APPROVED** [review]

**Comment by villebro** [comment]
@audour this PR has a conflict - can you rebase it so we can get these comprehensive fixes merged? Thanks!

**Comment by audour** [comment]
> @audour this PR has a conflict - can you rebase it so we can get these comprehensive fixes merged? Thanks!

Hi @villebro,
As I'm not very  familiar with Github suggestions (as proposed above by @mathieudruart and @martin-raymond), can you confirm that following steps would be OK:
1) accept manually each suggestion using "Commit suggestion" button => it will create as many commits as suggestions
2) do the rebase following this approach on the master branch latest commit : https://gist.github.com/scottyhq/299e4d36018a2f13acfb2528a1553002
Thanks in advance for your help

**Comment by villebro** [comment]
@audour I think this will work just fine. The PR will be squashed to a single commit during merging, so it doesn't really matter how many commits there are in the PR. So to reiterate what I think is most sensible here:
1. click through all suggestions.
2. one done, merge master on the PR, resolve the outstanding conflicts, then push and tag a committer so we can get it merged.

File: superset/translations/fr/LC_MESSAGES/messages.po [hunk 1]
```diff
@@ -1512,510 +1664,1103 @@ msgstr "Metadonnées rafraichies pour le(s) table(s) suivante(s) : %(tables)s"
...
 msgstr ""
+"Chaîne de connexion invalide, une chaîne valide a généralement cette forme : "
+"driver://user:password@database-host/database-name"
 
-#: superset/databases/schemas.py:181 superset/databases/schemas.py:196
+#: superset/databases/schemas.py:183 superset/databases/schemas.py:198
 #, python-format
 msgid "Field cannot be decoded by JSON. %(msg)s"
-msgstr ""
+msgstr "Le champ ne peut pas être décodé par JSON. %(msg)s"
```
[the remark this answers is not in the document]
Comment (audour) [reply superset/translations/fr/LC_MESSAGES/messages.po:1751]: I think that "par JSON" is more in line with english version "by JSON". Maybe the issue is more about the english version which could be "into JSON" ...

File: superset/translations/fr/LC_MESSAGES/messages.po [hunk 2]
```diff
@@ -1512,510 +1664,1103 @@ msgstr "Metadonnées rafraichies pour le(s) table(s) suivante(s) : %(tables)s"
...
 msgid "Connection failed, please check your connection settings"
-msgstr ""
+msgstr "La connexion a échoué, veuillez vérifier vos paramètres de connexion"
 
-#: superset/databases/commands/exceptions.py:108
-msgid "Cannot delete a database that has tables attached"
-msgstr ""
+#: superset/databases/commands/exceptions.py:109
+msgid "Cannot delete a database that has datasets attached"
+msgstr "Impossible de supprimer une base de données qui a des jeux de données "
```
[the remark this answers is not in the document]
Comment (audour) [reply superset/translations/fr/LC_MESSAGES/messages.po:1823]: I think we should keep a unique "dataset" translation in whole file. As there are 47 occurences for "jeu de données", in the whole file, I do not commit this specific suggestion.

File: superset/translations/fr/LC_MESSAGES/messages.po [hunk 3]
```diff
@@ -1512,510 +1664,1103 @@ msgstr "Metadonnées rafraichies pour le(s) table(s) suivante(s) : %(tables)s"
...
+msgstr ""
+"Dépassement de délai lors de la génération d'un CSV"
+
+#: superset/reports/commands/exceptions.py:175
+msgid "Alert fired during grace period."
+msgstr "Alerte déclenchée pendant la période de grace"
+
+#: superset/reports/commands/exceptions.py:179
+msgid "Alert ended grace period."
+msgstr "Alerte terminée grace période"
```
[the remark this answers is not in the document]
Comment (audour) [reply superset/translations/fr/LC_MESSAGES/messages.po]: Thanks for suggestions: fixed by https://github.com/apache/superset/pull/16868/commits/ece10a081522adf02b9e3daa512998a32005d75e

but I've translated differently as "Alert" seems to be the subject within the sentence for "to end" verb

**Comment by audour** [comment]
Thanks to @martin-raymond and @mathieudruart for their suggestions: I've applied almost all of them. For others I've answered with a comment.
@villebro: it seems that conflicts are resolved, I hope it's now ok to merge.

**Comment by junlincc** [comment]
@audour Thank you for the contribution! We can merge, as soon as the the checks pass :)

**Comment by villebro** [comment]
Thanks @audour - I kicked off the tests, will babysit this and ping you if there's anything that needs further attention.

**Comment by villebro** [comment]
@audour it appears you need to fix some lint to get CI green: 
image

**Comment by audour** [comment]
> @audour it appears you need to fix some lint to get CI green

Hi @villebro, sorry for that I've just pushed an update. News since last push:

* open fr PO file with Poedit which :
  * highlighted a typo error -> fixed
  * highlighted about a hundred suggestions about adding/removing trailing spaces, final dots or first capital letter -> fixed
* re-extract strings with `pybabel extract` to get new strings since I've rebased yesterday the working branch on `master` and re-update `fr` language file
* add missing strings and review all lines marked as "fuzzy" by POedit
* update the `.json`
* last check: re-run with success `pybabel extract`/`pybabel update`/`po2json.sh`/`pybabel compile`

Hope CI will turn green

**Comment by villebro** [comment]
Thanks so much for this! I started tests, fingers crossed 🤞

**Comment by villebro** [comment]
> @villebro I am curious why one of the tests is not passing as this PR seems super low risk?

We have a linting rule that expects the code to be formatted in a certain way. For instance, it doesn't allow for blank spaces at the end of the line or missing/excessive line changes at the end of files.

@audour do you mind taking another look? If needed I can help fix the issue.

**Comment by audour** [comment]
> @audour do you mind taking another look? If needed I can help fix the issue.

Hi @villebro, I can take another look but to be efficient I need to know the list of linting rules as the file `superset/translations/fr/LC_MESSAGES/messages.po` seems OK for Poedit and `pybabel` and to figure out how to find the issues locally without a dedicated linter.

**Comment by villebro** [comment]
This appears to be the violation:
```patch
diff --git a/superset/translations/fr/LC_MESSAGES/messages.po b/superset/translations/fr/LC_MESSAGES/messages.po
index c2058ee9e..8beec739c 100644
--- a/superset/translations/fr/LC_MESSAGES/messages.po
+++ b/superset/translations/fr/LC_MESSAGES/messages.po
@@ -10632,4 +10632,3 @@ msgstr "Zone de texte"
 #: superset-frontend/src/visualizations/TimeTable/TimeTableChartPlugin.js:36
 msgid "Trend"
 msgstr "Tendance"
-
```

**Comment by audour** [comment]
> This appears to be the violation (redundant line change at the end of the file):

Thanks, so is it ok with this new commit https://github.com/apache/superset/pull/16868/commits/b4a681349ab7bd5efa8359668a50f76ef7798d89 ?

**Comment by villebro** [comment]
> > This appears to be the violation (redundant line change at the end of the file):
> 
> Thanks, so is it ok with this new commit [b4a6813](https://github.com/apache/superset/commit/b4a681349ab7bd5efa8359668a50f76ef7798d89) ?

Yes, this seems to have fixed it 👍  merging as soon as the flaky test has passed (restarted now)

**Pull request merged**